File: app/src/test/kotlin/com/softyorch/stroopoverload/StroopStimulusTest.kt
package com.softyorch.stroopoverload

import com.softyorch.stroopoverload.core.StroopColor
import com.softyorch.stroopoverload.domain.StroopStimulus
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class StroopStimulusTest {

    @Test
    fun `correctAnswer is always inkColor`() {
        val s = StroopStimulus(wordLabel = StroopColor.RED, inkColor = StroopColor.BLUE)
        assertEquals(StroopColor.BLUE, s.correctAnswer)
    }

    @Test
    fun `isIncongruent true when wordLabel differs from inkColor`() {
        val s = StroopStimulus(wordLabel = StroopColor.RED, inkColor = StroopColor.GREEN)
        assertTrue(s.isIncongruent)
    }

    @Test
    fun `isIncongruent false when wordLabel equals inkColor`() {
        val s = StroopStimulus(wordLabel = StroopColor.RED, inkColor = StroopColor.RED)
        assertFalse(s.isIncongruent)
    }
}

File: test/domain/stroop_stimulus_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:stroop_overload/core/stroop_color.dart';
import 'package:stroop_overload/domain/stroop_stimulus.dart';

void main() {
  group('StroopStimulus', () {
    test('correctAnswer always equals inkColor', () {
      final s = StroopStimulus(
        wordLabel: StroopColor.red,
        inkColor: StroopColor.blue,
      );
      expect(s.correctAnswer, StroopColor.blue);
    });

    test('isIncongruent is true when wordLabel differs from inkColor', () {
      final s = StroopStimulus(
        wordLabel: StroopColor.red,
        inkColor: StroopColor.green,
      );
      expect(s.isIncongruent, isTrue);
    });

    test('isIncongruent is false when wordLabel equals inkColor', () {
      final s = StroopStimulus(
        wordLabel: StroopColor.red,
        inkColor: StroopColor.red,
      );
      expect(s.isIncongruent, isFalse);
    });
  });
}
